src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1973,8 +1978,31 @@ internal async Task ExecuteInternalCommand(FrontendMessage message, bool async)
 
         #region Misc
 
-        void HandleParameterStatus(string name, string value)
+        void ReadParameterStatus(ReadOnlySpan<byte> incomingName, ReadOnlySpan<byte> incomingValue)
         {
+            (byte[], byte[]) entry = default;
+            foreach (var e in _rawParameters)
+            {
+                if (incomingName.SequenceEqual(new ReadOnlySpan<byte>(e.Name)))
+                {
+                    entry = (e.Name, e.Value);
+                    break;
+                }
+            }
+
+            if (entry == default((byte[], byte[])))
+            {
+                entry = (incomingName.ToArray(), incomingValue.ToArray());
+                _rawParameters.Add(entry);
+            }
+            else if (!incomingValue.SequenceEqual(new ReadOnlySpan<byte>(entry.Item2)))
+                entry.Item2 = incomingValue.ToArray();
+            else
+                return;
+
+            var name = TextEncoding.GetString(entry.Item1);
+            var value = TextEncoding.GetString(entry.Item2);
+
```

Comment:
This stuff can be simplified.

```csharp
byte[] rawName;
byte[] rawValue;

foreach ((var currentName, var currentValue) in _rawParameters)
    if (incomingName.SequenceEqual(curentName))
    {
        if (incomingValue.SequenceEqual(currentValue))
            return;

        rawName = currentName;
        rawValue = incomingValue.ToArray();
        goto ProcessParameter;
    }

rawName = incomingName.ToArray();
rawValue = incomingValue.ToArray();
_rawParameters.Add((rawName, rawValue));

ProcessParameter:
var name = TextEncoding.GetString(rawName);
var value = TextEncoding.GetString(rawValue);
```

Reply:
Yeah, I tend to overlook goto :) Will integrate your changes (BTW you can use the new "code suggestions" feature in github, they're nice).

Reply:
(note that you can extract the two vars out of the parenthese in the foreach)

Reply:
Sometimes I use the feature, but as I know it doesn't allow to change multiple lines of code. Did I miss something?